File: archive/#1-twoSum/Solution.js
//
// Created by Penguin Zhang on 2021/3/7.
//
// MEMORY 38 MB/52.36%
// TIME 68 ms/99.07%

var twoSum = function(nums, target) {
    let n = nums.length;
    for(let i = 0; i < n; ++i)
        for(let j = i + 1; j < n; ++j)
            if(nums[i] + nums[j] === target)
                return [i, j];
    return [];
};

File: archive/#1-twoSum/Solution.c
//
// Created by Penguin Zhang on 2021/3/7.
//
// MEMORY 5.5 MB/100.00%
// TIME 4 ms/97.53%

int *twoSum(int *nums, int numsSize, int target, int *returnSize) {
    for (int i = 0; i < numsSize; ++i) {
        for (int j = i + 1; j < numsSize; ++j) {
            if (nums[i] + nums[j] == target) {
                int *ret = malloc(sizeof(int) * 2);
                ret[0] = i, ret[1] = j;
                *returnSize = 2;
                return ret;
            }
        }
    }
    *returnSize = 0;
    return NULL;
}